core/src/main/java/org/apache/cxf/common/spi/GeneratedNamespaceClassLoader.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.cxf.common.spi;
+
+import java.util.Map;
+
+public class GeneratedNamespaceClassLoader implements NamespaceClassCreator {
+    ClassLoader cl;
+    GeneratedNamespaceClassLoader(ClassLoader cl) {
+        this.cl = cl;
+    }
+    public synchronized Class<?> createNamespaceWrapper(Class<?> mcls, Map<String, String> map) {
```

Comment:
If I am not mistaken, `ClassLoader` already use synchronization for loading classes (@rmannibucau please correct me if I am wrong), why do you need  `synchronized` here?

Reply:
to avoid to generate twice the same class and register only once, +-0 to drop it

Reply:
I have removed it then think that if I switch to ClassLoaderUtils.loadClass I will have an issue because it can be thread relative whereas we need that class is generated once then discover whatever the thread (request received). So not sure it is goot to used it. We need a single classloader by called class not by thread. Does it make sens to you ?

Reply:
Let's keep it, I think we could implement simpler mechanism but not in scope of this PR